components/service-operator/apis/storage/v1beta1/s3_types_test.go

```diff
@@ -99,6 +101,14 @@ var _ = Describe("S3Bucket", func() {
 		Expect(o2).ToNot(BeZero())
 	})
 
+	It("should call the AWS API when emptied", func() {
+		client = &sdkfakes.FakeClient{}
+		ctx, _ = context.WithTimeout(context.Background(), time.Millisecond * 100)
```

Comment:
context leak! albeit in the tests :) 

you _must_ call `cancel` ... I think in this case you don't really need to create a new context ... you can just use `context.Background()` unless you are testing the timeout.